x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/prebuilt_rule_customization.md

```diff
@@ -428,4 +523,142 @@ And additionally this request contains one or more custom rules
+When a user reverts the rule
+Then the rule should be modified to match the original Elastic rule object corresponding to its version number
+And the rule's `is_customized` value should be false
+```
+
+#### **Scenario: User can view diff between current and original Elastic rule versions in flyout**
+
+**Automation**: 1 cypress test.
+
+```Gherkin
+Given a space with at least one prebuilt rule
+And that rule is customized
+And that rule has an existing base version
+When a user navigates to that rule's details page
+And clicks the overflow actions button
+And clicks the revert rule button
+Then a rule diff flyout should open
+And this flyout should be read-only
+And should list all fields that are different between the current and base version
+And should contain a button to revert the rule
+```
+
+#### **Scenario: Revert prebuilt rule button should be disabled when rule's base version is missing**
+
+**Automation**: 1 cypress test.
+
+```Gherkin
+Given a space with at least one prebuilt rule
+And that rule is customized
+And that rule does not have an existing base version
+When a user navigates to that rule's details page
+And clicks the overflow actions button
+Then the revert rule button should be disabled
+And have an informational tooltip on hover
+```
+
+#### **Scenario: Revert prebuilt rule button shouldn't appear if rule is non-customzied**
+
+**Automation**: 1 cypress test.
+
+```Gherkin
+Given a space with at least one prebuilt rule
+And that rule is non-customized
+When a user navigates to that rule's details page
+And clicks the overflow actions button
+Then the revert rule button should not be displayed as an option
+```
+
+#### **Scenario: Revert prebuilt rule endpoint returns error if rule's base version is missing**
+
+**Automation**: 1 integration test.
+
+```Gherkin
+Given a space with at least one prebuilt rule
+And that rule is customized
+And that rule does not have an existing base version
+When a user reverts the rule
+Then the API should return a 500 error
+And the rule should remain the same
+```
+
+#### **Scenario: Revert prebuilt rule endpoint returns error if request `revision` field doesn't match the rule associated with `rule_id`**
+
+**Automation**: 1 integration test.
+
+```Gherkin
+Given a space with at least one prebuilt rule
+And that rule is customized
+And that rule has an existing base version
+When a user reverts the rule with an outdated revision field
+Then the API should return a 500 error
+And the rule should remain the same
+```
+
+#### **Scenario: Revert prebuilt rule endpoint returns error if request `version` field doesn't match the rule associated with `rule_id`**
+
+**Automation**: 1 integration test.
+
+```Gherkin
+Given a space with at least one prebuilt rule
+And that rule is customized
+And that rule has an existing base version
+When a user reverts the rule with an outdated version field
+Then the API should return a 500 error
+And the rule should remain the same
+```
```

Comment:
These test scenarios could be under `Concurrency control` sub section. A name like `Reverting a rule to stock version: Concurrency control` may work.